src/link/Link.ts

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { createComponent, createHook } from "reakit-system";
+import { ClickableHTMLProps, ClickableOptions, useClickable } from "reakit";
+
+import { useForkRef } from "reakit-utils";
+import { useWarning } from "reakit-warning";
+
+export type LinkOptions = ClickableOptions & {
+  isExternal?: boolean;
+};
+
+export type LinkHTMLProps = ClickableHTMLProps;
+
+export type LinkProps = LinkOptions & LinkHTMLProps;
+
+export const useLink = createHook<LinkOptions, LinkHTMLProps>({
+  name: "Link",
+  compose: useClickable,
+  keys: ["isExternal"],
+
+  useOptions(options) {
+    return { unstable_clickOnSpace: false, ...options };
+  },
+
+  useProps({ isExternal }, { ref: htmlRef, ...htmlProps }) {
+    const ref = React.useRef<HTMLElement>(null);
+    const [role, setRole] = React.useState<"link" | undefined>(undefined);
```

Comment:
If they want link to behave as a button they can use `Button`

But, this is also possible, if you set `role="button"`, that will take the precedence.

Thats, not needed IMO, since we are not destructuring the role, when user gives a role, that will be returned.

You can achieve what you are asking with this snippet.
```
<Link as="button" role="button" onClick={() => alert("Custom Link")}>
      Reakit
    </Link>
```

Reply:
Yeah thats what i was saying, okay got it now!